fe/fe-core/src/main/java/org/apache/doris/datasource/iceberg/IcebergExternalTable.java

```diff
@@ -90,4 +188,200 @@ public long fetchRowCount() {
+            || DAY.equals(transformName)
+            || HOUR.equals(transformName);
+    }
+
+    protected IcebergPartitionInfo loadPartitionInfo(List<Column> partitionColumns,
+                                                     PartitionSpec spec, String transform) throws AnalysisException {
+        List<IcebergPartition> icebergPartitions = loadIcebergPartition(spec);
+        Map<String, IcebergPartition> nameToPartition = Maps.newHashMap();
+        Map<String, PartitionItem> nameToPartitionItem = Maps.newHashMap();
+        List<org.apache.doris.catalog.Type> types
+                = partitionColumns.stream().map(Column::getType).collect(Collectors.toList());
+        for (IcebergPartition partition : icebergPartitions) {
+            nameToPartition.put(partition.getPartitionValue(), partition);
+            PartitionItem item;
+            if (transform.equals(IDENTITY)) {
+                List<PartitionValue> values = Lists.newArrayList();
+                values.add(new PartitionValue(partition.getPartitionValue()));
+                PartitionKey key = PartitionKey.createListPartitionKeyWithTypes(values, types, true);
+                item = new ListPartitionItem(Lists.newArrayList(key));
+            } else {
+                Range<PartitionKey> partitionRange = getPartitionRange(Long.parseLong(partition.getPartitionValue()),
+                        transform, partitionColumns);
+                item = new RangePartitionItem(partitionRange);
+            }
+            nameToPartitionItem.put(partition.getPartitionValue(), item);
+        }
+        return new IcebergPartitionInfo(nameToPartitionItem, nameToPartition);
+    }
+
+    protected List<IcebergPartition> loadIcebergPartition(PartitionSpec spec) {
+        PartitionsTable partitionsTable = (PartitionsTable) MetadataTableUtils
+                .createMetadataTableInstance(table, MetadataTableType.PARTITIONS);
+        List<IcebergPartition> partitions = Lists.newArrayList();
+        Class<?> c = spec.javaClasses()[0];
+        try (CloseableIterable<FileScanTask> tasks = partitionsTable.newScan().planFiles()) {
+            for (FileScanTask task : tasks) {
+                // row format :
+                // 0. partitionData,
+                // 1. spec_id,
+                // 2. record_count,
+                // 3. file_count,
+                // 4. total_data_file_size_in_bytes,
+                // 5. position_delete_record_count,
+                // 6. position_delete_file_count,
+                // 7. equality_delete_record_count,
+                // 8. equality_delete_file_count,
+                // 9. last_updated_at,
+                // 10. last_updated_snapshot_id
+                CloseableIterable<StructLike> rows = task.asDataTask().rows();
+                for (StructLike row : rows) {
+                    StructProjection partitionData = row.get(0, StructProjection.class);
+                    String partitionValue = partitionData.get(0, c).toString();
```

Comment:
need to use the partition spec corresponding to the spec id to generate the correct partition information.
use `org.apache.iceberg.Table#specs` to get all specs